Source/Evolve/WarpXEvolveEM.cpp

```diff
@@ -270,7 +270,7 @@ WarpX::EvolveEM (int numsteps)
 
     bool write_plot_file = plot_int > 0 && istep[0] > last_plot_file_step
         && (max_time_reached || istep[0] >= max_step);
-    bool write_openPMD = openpmd_int > 0 && (max_time_reached || istep[0] >= max_step);
+    bool write_openPMD = (openpmd_int > 0) && (istep[0] > last_plot_file_step) && (max_time_reached || istep[0] >= max_step);
```

Comment:
hm, should the `last_plot_file_step` not only control AMReX plotfiles and not openPMD output?

Reply:
you mean last_plot_file_step only controls AMRex plot files? and another for openPMD plots? 
 
by the way,  what is  max_time_reached?  and what do you do when it is reached? and  when will (step[0] > max_step)?

Reply:
Yes, I would like to just add alongside the `last_plot_file_step` variable a `last_openPMD_step` variable, since both are independent I/O options.

I have no clue when `(step[0] > max_step)` might be true either and just kept the old logic. Maybe @MaxThevenet knows? We could check the git history who wrote the line for plotfiles.